.github/workflows/main.yaml

```
@@ -62,14 +62,28 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           fail_ci_if_error: false
+  publish:
+    runs-on: ubuntu-latest
```

```suggestion
    runs-on: ubuntu-latest
    timeout-minutes: 30
```
let's put some limit here too